### ISSUE TITLE
buildah is unable to push an image for a non-default platform 

### ISSUE THREAD

**Issue by santiagorr** [containers/buildah#5637]
If I build an image for a non-default platform, e.g. i386 on amd64, armhf or armel on arm64, without building the image for the default platform, buildah is not able to push it. It complains because it doesn't find the default platform-image:

```
buildah push build-essential <registry>
Error: pushing image "build-essential" to "<registry>": no image found in image index for architecture amd64, variant "", OS linux: image not known
```

I can reproduce it locally with a simple Dockerfile:

FROM debian:unstable
ARG ARCH=amd64
RUN apt-get update
RUN apt-get -y install build-essential

and building it with:

```
buildah build --build-arg ARCH=i386 --platform "linux/i386" --manifest build-essential
```

The error comes from `chooseInstance()`: https://github.com/containers/image/blob/main/internal/manifest/oci_index.go#L264

Building the default platform additionally fixes the issue (I am able to push the image), but I need to avoid that.

**Comment by nalind** [comment]
Yes, either `buildah manifest push --all` or `podman push` should push both the manifest list and the image.

**Comment by santiagorr** [comment]
Thanks for the answers to both of you @mtrmac, @nalind!

> Yes, either `buildah manifest push --all` or `podman push` should push both the manifest list and the image.

You are right, both commands are able to push the single-non-default-platform-image, but `buildah push ...` or `buidah push --all ...` don't.

**Comment by rhatdan** [comment]
@nalind Should we close this or better document the current behaviour?

**Comment by nalind** [comment]
Either the latter, or tweak how `push` works.
